packages/suite/src/hooks/wallet/useSendFormFields.ts

```diff
@@ -11,86 +17,147 @@ import { SendContextValues, UseSendFormState } from 'src/types/wallet/sendForm';
+    currencyCode: FiatCurrencyCode;
+    tokenAddress: TokenAddress;
+};
+
+type UseSendFormFieldsParams = UseFormReturn<FormState> & {
     fiatRate?: Rate;
     network: UseSendFormState['network'];
 };
 
 // This hook should be used only as a sub-hook of `useSendForm`
-
 export const useSendFormFields = ({
     getValues,
     setValue,
     clearErrors,
     network,
     formState: { errors },
-}: Props) => {
+}: UseSendFormFieldsParams) => {
     const { shouldSendInSats } = useBitcoinAmountUnit(network.symbol);
     const currentRates = useSelector(selectCurrentFiatRates);
 
-    const calculateFiat = useCallback(
-        (outputIndex: number, amount?: string) => {
-            const outputError = errors.outputs ? errors.outputs[outputIndex] : undefined;
-            const error = outputError ? outputError.amount : undefined;
+    const getCurrentFiatRate = useCallback(
+        ({ currencyCode, tokenAddress }: GetCurrentRateParams) => {
+            const fiatRateKey = getFiatRateKey(network.symbol, currencyCode, tokenAddress);
 
-            if (error) {
-                amount = undefined;
-            }
+            return currentRates?.[fiatRateKey];
+        },
+        [currentRates, network.symbol],
+    );
 
+    type CalculateFiatFromAmountOrViceVersaParams = {
+        outputId: number;
+        target: 'fiat' | 'amount';
+        formatTargetValue: (value: string, fiatRate: number) => string | null;
+        value?: string;
+    };
+
+    const calculateFiatFromAmountOrViceVersa = useCallback(
+        ({
+            formatTargetValue,
+            outputId,
+            target,
+            value,
+        }: CalculateFiatFromAmountOrViceVersaParams) => {
             const { outputs } = getValues();
-            const output = outputs ? outputs[outputIndex] : undefined;
-            if (!output || output.type !== 'payment') return;
-            const { fiat, token, currency } = output;
-            if (typeof fiat !== 'string') return; // fiat input not registered (testnet or fiat not available)
-            const inputName = `outputs.${outputIndex}.fiat` as const;
-            if (!amount) {
-                // reset fiat value (Amount field has error)
-                if (fiat.length > 0) {
-                    setValue(inputName, '');
+            const output = outputs[outputId];
+            if (output.type !== 'payment') {
+                return;
+            }
+            const targetValue = output[target];
+            if (target === 'fiat' && typeof targetValue !== 'string') {
+                return; // fiat input not registered (testnet or fiat not available)
+            }
+            const targetInputName = `outputs.${outputId}.${target}` as const;
+            const outputError = errors.outputs ? errors.outputs[outputId] : undefined;
+            const error = outputError
+                ? outputError[target === 'fiat' ? 'amount' : 'fiat']
+                : undefined;
+            if (error || !value) {
```

Comment:
Here's what fixes the bug. In `FiatInput`, `error` was read from context where it was not yet updated at the time of triggering the change handler. Thus, this behaviour was delayed by one keyboard stroke.
Accessing `error` directly from `useForm` solves the issue. `Amount` did not have this problem, so I just unified the approach between the two inputs.